Packs/CrowdStrikeFalconStreamingV2/Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py

```diff
@@ -149,69 +143,78 @@ async def fetch_event(
-                                    event_metadata = streaming_event.get('metadata', {})
-                                    event_creation_time = event_metadata.get('eventCreationTime', 0)
-                                    if not event_creation_time:
-                                        demisto.debug('Could not extract "eventCreationTime" field, using 0 instead. '
-                                                      f'{streaming_event}')
-                                    else:
-                                        event_creation_time /= 1000
-                                    event_creation_time_dt = datetime.fromtimestamp(event_creation_time)
-                                    if event_creation_time_dt < first_fetch_time:
-                                        demisto.debug(f'Event with offset {event_metadata.get("offset")} '
-                                                      f'and creation time {event_creation_time} was skipped.')
-                                        continue
-                                    yield streaming_event
-                                except json.decoder.JSONDecodeError:
-                                    demisto.debug(f'Failed decoding event (skipping it) - {str(stripped_line)}')
-                            else:
-                                new_lines_fetched += 1
-                            if last_fetch_stats_print + timedelta(minutes=1) <= datetime.utcnow():
-                                demisto.info(
-                                    f'Fetched {events_fetched} events and'
-                                    f' {new_lines_fetched} new lines'
-                                    f' from the stream in the last minute.')
-                                events_fetched = 0
-                                new_lines_fetched = 0
-                                last_fetch_stats_print = datetime.utcnow()
-                except Exception as e:
-                    demisto.debug(f'Failed to fetch event: {e} - Going to sleep for 10 seconds and then retry -'
-                                  f' {traceback.format_exc()}')
-                    await sleep(10)
+            try:
+                event = Event()
+                create_task(self._discover_refresh_stream(event))
+                demisto.debug('Waiting for stream discovery')
+                await wait_for(event.wait(), 30)
+            except TimeoutError as e:
+                demisto.debug(f'Failed discovering stream: {e} - '
+                              f'Going to sleep for 30 seconds and then retry - {traceback.format_exc()}')
+                await sleep(30)
+            else:
+                demisto.debug('Done waiting for stream discovery')
+                events_fetched = 0
+                new_lines_fetched = 0
+                last_fetch_stats_print = datetime.utcnow()
+                client_timeout = MINUTES_25 + uniform(-5 * 60, 5 * 60)
```

Comment:
I would say lets random between 15-25 minutes to be on the safe side, and not 20-30. 30 is the maximum limit.